.github/workflows/pypy_build_publish_template.yaml

```diff
@@ -0,0 +1,38 @@
+# Reusable workflow to setup a specific WMCore component for pip
+
+on:
+  workflow_call:
+    inputs:
+      wmcore_component:
+        required: true
+        type: string
+    secrets:
+      pypy_token:
+        required: true
+
+jobs:
+  build_and_publish_from_template:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup python 3.8
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.8"
+      - name: Upgrade pip3
+        run: |
+          python3 -m pip install --upgrade pip
+      - name: Update the setup script template with package name
+        run: |
+          sed "s/PACKAGE_TO_BUILD/${{ inputs.wmcore_component }}/" setup_template.py > setup.py
+      - name: Create requirements file
+        run: |
+          cp requirements.txt requirements.wmcore.txt
+          awk "/(${{ inputs.wmcore_component }}$)|(${{ inputs.wmcore_component }},)/ {print \$1}" requirements.wmcore.txt > requirements.txt
+      - name: Build sdist
+        run: python setup.py clean sdist
+      - name: Publish component
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.pypy_token }}
```

Comment:
Is it supposed to be the same secrets as the yaml file above? I see they refer to different variables, is it correct?

Reply:
@amaltaro , the GitHub Actions does not run on RPM based systems, that's why we are trying to convert to pip/pypi. And, if we are doing this right it should not be matter since python code should run identically on RHAL and Debian using pypi/pip approach.

Reply:
@amaltaro I have squashed the commits. Regarding Ubuntu: We are not requiring any packages other than python 3.8 at buld-time for pypy, so I think it doesn't really make any difference.

Reply:
> Is it supposed to be the same secrets as the yaml file above? I see they refer to different variables, is it correct?

Yes. The template has its own variable name (`pypy_token`). The other file, which is the action itself, makes sure that we map `pypy_token` to the token name defined on the github settings for actions